recipes/php/php.inc

```diff
@@ -0,0 +1,85 @@
+DESCRIPTION = "PHP is a popular general-purpose scripting language that is especially suited to web development."
+LICENSE = "PHP-3.01"
+HOMEPAGE = "http://php.net"
+
+COMPATIBLE_HOST_ARCHS = ".*linux"
+RECIPE_TYPES = "native machine"
+
+inherit autotools c c++ pkgconfig
+
+SRC_URI = "http://in1.php.net/distributions/php-${PV}.tar.xz"
+
+PHP_LIBDIR = "${libdir}/php5"
+
+DEPENDS += "libc libm libcrypt"
+DEPENDS_${PN} += "libc libm libcrypt"
+RDEPENDS_${PN} += "libc libm libcrypt"
+
+
+EXTRA_OECONF = " \
+  --libdir=${PHP_LIBDIR} \
+  --with-libdir=${base_libdir} \
+  --disable-all \
+  --enable-json \
+  --enable-calendar \
+"
+
+EXTRA_OEMAKE = "INSTALL_ROOT=${D}"
+
+FILES_${PN}-dev += "${PHP_LIBDIR}/build"
+
+RECIPE_FLAGS += "php_ssl"
+DEFAULT_USE_php_ssl = True
+EXTRA_OECONF_SSL = "--without-openssl"
+EXTRA_OECONF_SSL:USE_php_ssl = "--with-openssl"
+EXTRA_OECONF += "${EXTRA_OECONF_SSL}"
+DEPENDS_SSL = ""
+DEPENDS_SSL:USE_php_ssl = "libssl libcrypto"
+DEPENDS += "${DEPENDS_SSL}"
+DEPENDS_${PN} += "${DEPENDS_SSL}"
+RDEPENDS_${PN} += "${DEPENDS_SSL}"
+
+RECIPE_FLAGS += "php_sockets"
+DEFAULT_USE_php_sockets = True
+EXTRA_OECONF_SOCKETS = "--disable-sockets"
+EXTRA_OECONF_SOCKETS:USE_php_sockets = "--enable-sockets"
+EXTRA_OECONF += "${EXTRA_OECONF_SOCKETS}"
+
+RECIPE_FLAGS += "php_session"
+DEFAULT_USE_php_session = True
+EXTRA_OECONF_SESSION = "--disable-session"
+EXTRA_OECONF_SESSION:USE_php_session = "--enable-session"
+EXTRA_OECONF += "${EXTRA_OECONF_SESSION}"
```

Comment:
Probably not, but there may be some who want to just use php as an interpreter on the target without using it with a webserver, so in that case it might make sense to disable the webby things. I actually almost had this is a use case, but that was then dropped for other reasons.